rs/nns/governance/src/proposals/execute_nns_function.rs

```diff
@@ -26,6 +35,56 @@ impl ValidExecuteNnsFunction {
     pub(crate) fn topic(&self) -> Topic {
         self.nns_function.topic()
     }
+
+    /// Converts the ExecuteNnsFunction to a self-describing value using the candid file fetched
+    /// from the management canister (as canister metadata).
+    async fn to_self_describing_value(
+        &self,
+        env: Arc<dyn Environment>,
+    ) -> Result<SelfDescribingValue, String> {
+        let candid_source = self.get_candid_source(env).await?;
+        let (_, method_name) = self.nns_function.canister_and_function();
+        candid_to_self_describing(&candid_source, method_name, &self.payload)
+    }
+
+    pub async fn to_self_describing_action(
+        &self,
+        env: Arc<dyn Environment>,
+    ) -> Result<SelfDescribingProposalAction, GovernanceError> {
+        let value = self.to_self_describing_value(env).await.map_err(|e| {
+            GovernanceError::new_with_message(
+                ErrorType::InvalidProposal,
+                format!(
+                    "Failed to convert ExecuteNnsFunction to self-describing value: {}",
+                    e
+                ),
+            )
+        })?;
+        Ok(SelfDescribingProposalAction {
+            type_name: self.nns_function.type_name().to_string(),
+            type_description: self.nns_function.type_description().to_string(),
+            value: Some(value),
```

Comment:
Perhaps, in a future PR,

```suggestion
            payload: Some(payload),
```